### PR TITLE
very basic api

### PR DESCRIPTION
**Pull request by afeena**
Api gives the possibility to get the list of the available SNARE uuids and get the info (sessions) of certain uuid.

### CODE REVIEW

**Comment by afeena** [comment]
I think about any frameworks for working with api, maybe Flask, maybe we can migrate to it, but if we will have very limited range of api functions, maybe it will be redundant

File: session_manager.py [hunk 1]
```diff
@@ -17,6 +17,8 @@ def add_or_update_session(self, raw_data):
         yield from self.delete_old_sessions()
         # handle raw data
         valid_data = self.validate_data(raw_data)
+        # push snare uuid into redis.
+        self.r.sadd(uuid.uuid3(uuid.NAMESPACE_DNS, 'snare_uuids').hex, valid_data['uuid'])
```
Comment (afeena) [inline session_manager.py]: Maybe we can use more suitable key for snare uuids

**Comment by glaslos** [comment]
I agree that Flask might be overkill. aiohttp should be good enough.

File: session_manager.py (hunk 1 above)
Comment (glaslos) [inline session_manager.py]: can't we just call the set `snare_ids`?

File: api.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        if path[-1] == '/':
+            path = path[:-1]
+
+        request, arg = os.path.split(path)
+
+        if request == '/api' and arg == 'stats':
+            result = yield from self.return_stats()
+        if request == '/api/stats':
+            result = yield from self.return_uuid_stats(uuid)
```
Comment (glaslos) [inline api.py]: I think you intended to do something else here.

File: session_manager.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import asyncio
 import redis
-
+import uuid
```
Comment (glaslos) [inline session_manager.py]: not required anymore

File: api.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        self.r = redis.StrictRedis(host='localhost', port=6379, decode_responses=True)
+
+    @asyncio.coroutine
+    def handle_api_request(self, path):
+        result = None
+
+        if path[-1] == '/':
+            path = path[:-1]
+
+        request, arg = os.path.split(path)
```
Comment (glaslos) [inline api.py]: I'd use `/api/stats` and `/api/stats?uuid=aaaa-bbbb....` and then https://docs.python.org/3/library/urllib.parse.html or aiohttp to parse the request into components.

**Pull request merged**